src/arc_spice/eval/inference_utils.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Callable
+
+import torch
+from sklearn.metrics import hamming_loss, zero_one_loss
+from torch.utils.data import DataLoader
+from tqdm import tqdm
+
+from arc_spice.data.multieurlex_utils import MultiHot
+from arc_spice.eval.translation_error import get_comet_model
+from arc_spice.variational_pipelines.RTC_single_component_pipeline import (
+    RTCSingleComponentPipeline,
+)
+from arc_spice.variational_pipelines.RTC_variational_pipeline import (
+    RTCVariationalPipeline,
+)
+
+
+class ResultsGetter:
+    def __init__(self, n_classes):
+        self.func_map: dict[str, Callable] = {
+            "recognition": self.recognition_results,
+            "translation": self.translation_results,
+            "classification": self.classification_results,
+        }
+        self.comet_model = get_comet_model()
+        self.multihot = MultiHot(n_classes)
+
+    def get_results(
```

Comment:
this is actually doing inference if I'm reading this correctly? imo current name suggests it's just fetching precalculated results

Reply:
oh but it calls functions from the func_map and I thought those performed inference?

Reply:
ah ok because `get_results` *updates* the results_dict. I hadn't realised that. I don't think it's worth changing here but, In future, I think we should avoid having those sort of side effects and where we do need to have them, make them super clear.

Reply:
Ah that's my fault, I should name this something else. This is the functions used to obtain results, not the forward methods themselves